modules/aleph/includes/aleph.reservation.inc

```diff
@@ -213,8 +215,57 @@ function aleph_reservation_list($account, $requested_type = NULL) {
   return $result;
 }
 
-function aleph_reservation_create() {
-  return array();
+/**
+ * Create a new reservation in the library system.
+ *
+ * @param object $account
+ *   Drupal user account object.
+ * @param string $id
+ *   ID of reservation.
+ *
+ * @return array
+ *   Reservation information.
+ *
+ * @throws DingProviderReservationExists
+ * @throws DingProviderReservationNotAllowed
+ * @throws DingProviderReservationNotAvailable
```

Comment:
Not all the exceptions thrown here seems to be thrown in practice e.g. `DingProviderReservationNotAllowed` and `DingProviderReservationNotAvailable`. Is that intentional?